File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/common/ImageTests.java
package com.epam.jdi.uitests.testing.unittests.tests.common;

import com.epam.jdi.uitests.core.interfaces.common.IImage;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication;
import com.epam.jdi.uitests.win.winium.elements.common.Image;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.HOME_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.homePage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.checkText;

/**
 * Created by Dmitry_Lebedev1 on 15/12/2015.
 */
public class ImageTests extends InitTests {
    private static final String ALT = "ALT";
    private static final String SRC = "http://ecse00100176.epam.com/label/Logo_Epam_Color.svg";
    private IImage clickableItem() { return homePage.logoImage; }

    @BeforeMethod
    public void before(final Method method) {
        isInState(HOME_PAGE, method);
    }

    @Test
    public void clickTest() throws InterruptedException {
        EpamJDIApplication.contactFormPage.open();
        clickableItem().click();
        EpamJDIApplication.homePage.checkOpened();
    }

    @Test
    public void setAttributeTest() {
        String _attributeName = "testAttr";
        String _value = "testValue";
        clickableItem().setAttribute(_attributeName, _value);
        checkText(() -> ((Image) clickableItem()).getWebElement().getAttribute(_attributeName), _value);
    }

    @Test
    public void getSourceTest() {
        checkText(clickableItem()::getSource, SRC);
    }

    @Test
    public void getTipTest() {
        checkText(clickableItem()::getAlt, ALT);
    }
}

File: Python/Test/jdi_uitests_webtests/test/common/image_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class ImageTextx(InitTests):
    ALT = "ALT"
    SRC = "https://jdi-framework.github.io/tests/images/Logo_Epam_Color.svg"

    clickable_item = EpamJDISite.home_page.logo_image

    def setUp(self):
        Preconditions.HOME_PAGE.is_in_state()

    def test_click(self):
        EpamJDISite.contact_form_page.open()
        self.clickable_item.click()
        EpamJDISite.home_page.check_opened()

    def test_set_attribute(self):
        attribute_name = "testAttr"
        value = "testValue"
        self.clickable_item.set_attribute(attribute_name, value)
        Assert.check_text(self.clickable_item.get_element().get_attribute(attribute_name), value)

    def test_get_source(self):
        Assert.check_text(self.clickable_item.get_source(), self.SRC)

    def test_get_tip(self):
        Assert.check_text(self.clickable_item.get_alt(), self.ALT)